packages/wrangler/src/dev/start-server.ts

```diff
@@ -317,19 +319,19 @@ export async function startLocalServer(props: LocalProps) {
 	logger.log(chalk.dim("⎔ Starting local server..."));
 
 	const config = await localPropsToConfigBundle(props);
-	return new Promise<{ stop: () => void }>((resolve, reject) => {
+	return new Promise<{ stop: () => Promise<void> }>((resolve, reject) => {
 		const server = new MiniflareServer();
 		server.addEventListener("reloaded", async (event) => {
 			await maybeRegisterLocalWorker(event, props.name);
 			props.onReady?.(event.url.hostname, parseInt(event.url.port));
 			// Note `unstable_dev` doesn't do anything with the inspector URL yet
 			resolve({
-				stop: () => {
+				stop: async () => {
 					abortController.abort();
 					logger.log("⎔ Shutting down local server...");
 					// Initialization errors are also thrown asynchronously by dispose().
 					// The `addEventListener("error")` above should've caught them though.
-					server.onDispose().catch(() => {});
+					await server.onDispose();
 					removeMiniflareServerExitListener();
```

Comment:
Fallen in the trap of async cleanup. We should either call sync stuff first and async last
```suggestion
					removeMiniflareServerExitListener();
					await server.onDispose();
```

Or trigger async stuff but wait for it at the end of the function:
```suggestion
					const disposePromise = server.onDispose();
					removeMiniflareServerExitListener();
					return Promise.all([ disposePromise, ... ]);
```